sdl_android/src/main/java/com/smartdevicelink/streaming/RTPH264Packetizer.java

```diff
@@ -0,0 +1,488 @@
+
+	/**
+	 * Sets the Payload Type (PT) of RTP header field.
+	 *
+	 * Use this method if PT needs to be specified. The value should be between 0 and 127.
+	 * Otherwise, a default value (96) is used.
+	 *
+	 * @param type A value indicating the Payload Type
+	 */
+	public void setPayloadType(byte type) {
+		if (type >= 0 && type <= 127) {
+			mPayloadType = type;
+		} else {
+			mPayloadType = DEFAULT_RTP_PAYLOAD_TYPE;
+		}
+	}
+
+	/**
+	 * Sets the SSRC of RTP header field.
+	 *
+	 * Use this method if SSRC needs to be specified. Otherwise, a random value is generated and
+	 * used.
+	 *
+	 * @param ssrc An integer value representing SSRC
+	 */
+	public void setSSRC(int ssrc) {
+		mSSRC = ssrc;
+	}
+
+	/**
+	 * Overwrites MTU value. This is only for testing.
+	 *
+	 * @param mtu maximum payload size of SDL frames
+	 */
+	public void setMTU(int mtu) {
+		if (mtu <= 0) {
+			mtu = MAX_DATA_SIZE_FOR_ENCRYPTED_SERVICE;
+		}
+		bufferSize = mtu;
+	}
+
+	/**
+	 * Starts this packetizer.
+	 *
+	 * It is recommended that the video encoder is started after the packetizer is started.
+	 */
+	@Override
+	public void start() throws IOException {
+		if (mThread != null) {
+			return;
+		}
+
+		mThread = new Thread(this);
+		mThread.start();
+	}
+
+	/**
+	 * Stops this packetizer.
+	 *
+	 * It is recommended that the video encoder is stopped prior to the packetizer.
+	 */
+	@Override
+	public void stop() {
+		if (mThread == null) {
+			return;
+		}
+
+		mThread.interrupt();
+		mThread = null;
+
+		mPaused = false;
+		mWaitForIDR = false;
+		mOutputQueue.clear();
+	}
+
+	/**
+	 * Pauses this packetizer.
+	 *
+	 * This pauses the packetizer but does not pause the video encoder.
+	 */
+	@Override
+	public void pause() {
+		mPaused = true;
+	}
+
+	/**
+	 * Resumes this packetizer.
+	 */
+	@Override
+	public void resume() {
+		mWaitForIDR = true;
+		mPaused = false;
+	}
+
+	/**
+	 * The thread routine.
+	 */
+	public void run() {
+		SdlConnection connection = _session.getSdlConnection();
+
+		while (mThread != null && !mThread.isInterrupted()) {
+			ByteBuffer frame;
+			try {
+				frame = mOutputQueue.take();
+			} catch (InterruptedException e) {
+				Thread.currentThread().interrupt();
+				break;
+			}
+
+			while (frame.hasRemaining()) {
+				int len = frame.remaining() > bufferSize ? bufferSize : frame.remaining();
+
+				ProtocolMessage pm = new ProtocolMessage();
+				pm.setSessionID(_rpcSessionID);
+				pm.setSessionType(_serviceType);
+				pm.setFunctionID(0);
+				pm.setCorrID(0);
+				pm.setData(frame.array(), frame.arrayOffset() + frame.position(), len);
+				pm.setPayloadProtected(mServiceProtected);
+
+				_streamListener.sendStreamPacket(pm);
+
+				frame.position(frame.position() + len);
+			}
+		}
+
+		// XXX: This is added to sync with StreamPacketizer. Actually it shouldn't be here since
+		// it's confusing that a packetizer takes care of End Service request.
+		if (connection != null) {
+			connection.endService(_serviceType, _rpcSessionID);
```

Comment:
Agreed. This seems better as a callback from a packetizer to indicated the stream is empty/consumed and the implementing class should then be responsible for closing the service. Though it might be out of the scope for this PR and needs to be addressed in all packetizers.

Reply:
Yes, I prefer to make it out of the scope of this PR.